distributed/scheduler.py

```diff
@@ -271,6 +271,12 @@ def __init__(
 
         self.extra = extra or {}
 
+    def __hash__(self):
+        return hash((self.name, self.host))
+
+    def __eq__(self, other):
+        return type(self) == type(other) and hash(self) == hash(other)
```

Comment:
Sorry for being late to the party. There is a bug in here: It is not guaranteed that two WorkerState objects are actually the same even though there hash value is the same. Hashing is not a bijective function.

I think we need to go the extra route here and check for `(self.name, self.host) == (other.name, other.host)`

Reply:
@TomAugspurger  @bnaul  please see above.

While I totally agree that hash collisions are very very unlikely, Dask has a massive install base with sometimes hundreds to thousands of workers per cluster. So as far as we know the once in a million chance event be next Tuesday. 🤷‍♂